provider/src/tzif.rs

```diff
@@ -0,0 +1,155 @@
+//! A compact, zero copy TZif file.
+//!
+//! NOTE: This representation does not follow the TZif specification
+//! to full detail, but instead attempts to compress TZif data into
+//! a functional, data driven equivalent.
+
+#[cfg(feature = "datagen")]
+use alloc::vec::Vec;
+
+#[cfg(feature = "datagen")]
+use std::{collections::BTreeMap, path::Path};
+use zerotrie::ZeroAsciiIgnoreCaseTrie;
+#[cfg(feature = "datagen")]
+use zerotrie::ZeroTrieBuildError;
+use zerovec::{vecs::Index32, VarZeroVec, ZeroVec};
+#[cfg(feature = "datagen")]
+use zoneinfo_rs::{compiler::CompiledTransitions, ZoneInfoCompiler, ZoneInfoData};
+
+#[cfg(feature = "datagen")]
+use crate::tzdb::TzdbDataSource;
+
+#[derive(Debug, Clone)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(yoke::Yokeable, serde::Serialize, databake::Bake)
+)]
+#[cfg_attr(feature = "datagen", databake(path = timezone_provider::tzif))]
+pub struct ZoneInfoProvider<'data> {
+    // IANA identifier map to TZif index.
+    pub ids: ZeroAsciiIgnoreCaseTrie<ZeroVec<'data, u8>>,
+    // Vector of TZif data
+    pub tzifs: VarZeroVec<'data, ZeroTzifULE, Index32>,
+}
+
+#[zerovec::make_varule(ZeroTzifULE)]
+#[derive(PartialEq, Debug, Clone)]
+#[zerovec::skip_derive(Ord)]
+#[zerovec::derive(Debug)]
+#[cfg_attr(
+    feature = "datagen",
+    derive(yoke::Yokeable, serde::Serialize, databake::Bake)
+)]
+#[cfg_attr(feature = "datagen", zerovec::derive(Serialize))]
+#[cfg_attr(feature = "datagen", databake(path = timezone_provider::tzif))]
+pub struct ZeroTzif<'data> {
+    pub transitions: ZeroVec<'data, i64>,
+    pub transition_types: ZeroVec<'data, u8>,
+    // NOTE: zoneinfo64 does a fun little bitmap str
+    pub types: ZeroVec<'data, LocalTimeRecord>,
+    pub posix: &'data str,
```

Comment:
I would propose parsing the posix string, instead of leaving it as a raw string. IIRC, zoneinfo64 has the `rules` field, which could map nicely into a common rules format for tzif and zoneinfo64, right?

Reply:
I believe this is changed in #265. If it isn't then, it would be easy to add on that follow up. This was primarily a filler at the time.

Reply:
> zoneinfo64 has the `Rules` field, which could map nicely into a common rules format for tzif and zoneinfo64, right?

It does have a rules format, but I haven't looked super deeply at it or its purpose. It seemed a little strange or edge case based. The posix struct would map to the `final_*` fields in the zone table entry.